code/ship/ship.cpp

```diff
@@ -4125,34 +4885,34 @@ int get_default_player_ship_index()
 // Goober5000 - this works better in its own function
 void ship_set_default_player_ship()
 {
-	int i;
-
 	// already have one
 	if(strlen(default_player_ship))
 		return;
 
+	auto it = Ship_info.cbegin();
```

Comment:
As far as I am aware, it's preferable to declare a variable once and then reuse it (see also: every single function where "int i;" is declared at the beginning and then "i" is used as a loop iterator multiple times in a row, being reset each time). Is it actually preferable to declare a new variable every time we have a for loop? Or will the compiler optimize both so that the difference is irrelevant and it all comes down to which source reads better?